File: app/src/main/java/com/turlir/abakcalc/MainView.java
package com.turlir.abakcalc;

import android.content.Context;

import com.turlir.calculator.converter.Visual;

import java.util.List;

public interface MainView {

    void showResult(String digit);

    void showError(String error);

    Context context();

    void setRepresentation(List<CalculatorVisual> v);

    void setNotation(List<Visual> v);

    void resetToEmpty();
}

File: app/src/main/kotlin/com/turlir/abakcalc/MainView.kt
package com.turlir.abakcalc

import android.content.Context

import com.turlir.calculator.converter.Visual

interface MainView {

    fun showResult(digit: String)

    fun showError(error: String)

    val context: Context

    fun setRepresentation(v: List<CalculatorVisual>)

    fun setNotation(v: List<Visual>)

    fun resetToEmpty()
}
